### PR TITLE
Add q2

### PR DESCRIPTION
**Pull request by Fvergnet**
fix convergence order in space with one and two particles

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -7,6 +7,7 @@ include_directories(cafes external_packages/qhull)
 
 #set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++14")
 set (CMAKE_CXX_STANDARD 14)
+#set(DOLFIN_SKIP_BUILD_TESTS true)
```
Comment (gouarin) [inline CMakeLists.txt]: remove this line

File: .dockerignore [hunk 2]
```diff
@@ -0,0 +1,8 @@
+demos
```
Comment (gouarin) [inline .dockerignore]: .dockerignore is not needed. Please remove this file.

File: cafes/fem/assembling.hpp [hunk 3]
```diff
@@ -81,6 +81,10 @@ PetscErrorCode PreallocateMat(Context *user, Options opt, const MatType mtype,
     ierr = MatSetType(A, (const MatType)mtype);CHKERRQ(ierr);
     ierr = MatSetDM(A, pack);CHKERRQ(ierr);
     ierr = MatSetFromOptions(A);CHKERRQ(ierr);
+    // ierr = MatSetUp(A);CHKERRQ(ierr);
+    // ierr = MatSetOption(A, MAT_NEW_NONZERO_LOCATIONS, PETSC_TRUE);CHKERRQ(ierr);
+    // ierr = MatSetOption(A, MAT_NEW_NONZERO_LOCATION_ERR, PETSC_FALSE);CHKERRQ(ierr);
+    // ierr = MatSetOption(A, MAT_NEW_NONZERO_ALLOCATION_ERR, PETSC_FALSE);CHKERRQ(ierr);
```
Comment (gouarin) [inline cafes/fem/assembling.hpp]: Do we need to keep these lines ?

File: cafes/fem/assembling.hpp [hunk 4]
```diff
@@ -174,6 +178,11 @@ PetscErrorCode PreallocateMat(Context *user, Options opt, const MatType mtype,
     ierr = MatPreallocateFinalize(dnz, onz);CHKERRQ(ierr);
 
     ierr = MatSetLocalToGlobalMapping(A, ltog, ltog);CHKERRQ(ierr);
+
+    // ierr = MatSeqAIJSetPreallocation(A, PETSC_DEFAULT, PETSC_NULL);CHKERRQ(ierr);
+    // ierr = MatMPIAIJSetPreallocation(A, 0, PETSC_NULL, 0, PETSC_NULL);CHKERRQ(ierr);
+    // MatSetOption(A,MAT_NEW_NONZERO_LOCATION_ERR,PETSC_FALSE);
+
```
Comment (gouarin) [inline cafes/fem/assembling.hpp]: same here

File: cafes/fem/matElem.hpp [hunk 5]
```diff
@@ -141,6 +141,15 @@ std::vector<double> matElemMass_pu_2d(std::size_t order)
...
                 0, 0, 0, 1./9};
+        // return {1./9, 0, 0, 0,
+        //         0, 2./9, 0, 0,
+        //         1./9, -2./9, 0, 0,
+        //         0, 0, 2./9, 0,
+        //         0, 0, 0, 4./9,
+        //         0, 0, 2./9, -4./9,
+        //         1./9, 0, -2./9, -4./9,
+        //         0, 2./9, 0, -4./9,
+        //         1./9, -2./9, -2./9, 4./9};
```
Comment (gouarin) [inline cafes/fem/matElem.hpp]: same here

File: cafes/io/hdf5.hpp [hunk 6]
```diff
@@ -37,7 +37,7 @@
 #include <petscviewerhdf5.h>
 // #include <highfive/H5File.hpp>
 
-#include "../problem/particle_operator.hpp"
+// #include "../problem/particle_operator.hpp"
```
Comment (gouarin) [inline cafes/io/hdf5.hpp]: you can remove it. We will fix the singularity output after

File: cafes/problem/dton.hpp [hunk 7]
```diff
@@ -177,7 +160,7 @@ struct convergence_context2
             KSP ksp;
             PetscInt kspiter;
             PetscReal kspresnorm;
-            std::size_t scale_ = 4;
+            std::size_t scale_ = 1;
```
Comment (gouarin) [inline cafes/problem/dton.hpp:163]: Could you let 4 ?

File: cafes/problem/stokes.hpp [hunk 8]
```diff
@@ -338,7 +338,7 @@ namespace cafes
                 PetscFunctionBeginUser;
                 PC pc;
 
-                ierr = KSPCreate(PETSC_COMM_WORLD, &ksp);CHKERRQ(ierr);
+                ierr = KSPCreate(PETSC_COMM_WORLD, &ksp);//CHKERRQ(ierr);
```
Comment (gouarin) [inline cafes/problem/stokes.hpp]: remove the comment to check the petsc error

**Review by gouarin — CHANGES_REQUESTED** [review]

File: cafes/fem/assembling.hpp (hunk 3 above)
Comment (Fvergnet) [reply cafes/fem/assembling.hpp]: No, there is no need to keep them.

File: cafes/problem/dton.hpp (hunk 7 above)
Comment (Fvergnet) [reply cafes/problem/dton.hpp:163]: Well, I do not know. I do not see the point to let 4 if it works with 1. But may be there are some particular cases where we need more points on the surface of the particles. What was the reason to put 4 in the first place ?

Comment (gouarin) [reply cafes/problem/dton.hpp:163]: let 1, then !

**Comment by gouarin** [comment]
Thanks @Fvergnet  !

**Pull request merged**